lib/debug_socket.rb

```diff
@@ -16,7 +17,10 @@ def self.logger
   end
 
   def self.start(path)
-    raise "debug socket thread already running" if @thread
+    pid = Process.pid
+    raise "debug socket thread already running for this process" if @thread && @pid == pid
```

Comment:
When MRI forks, it kills all other threads, so if process-A starts a debug socket, then forks, the new process-B child will not have a running `@thread`.